reactor3/src/test/java/com/linecorp/armeria/common/reactor3/RequestContextPropagationFluxTest.java

```diff
@@ -0,0 +1,466 @@
+        final Flux<String> flux;
+
+        final ConnectableFlux<String> connectableFlux = Flux.just("foo").publish();
+        flux = addCallbacks(connectableFlux.autoConnect(2, disposable -> {
+            assertThat(ctxExists(ctx)).isTrue();
+        }).publishOn(Schedulers.newSingle("aaa")), ctx);
+
+        final Disposable disposable1 = flux.subscribe();
+        await().pollDelay(Duration.ofMillis(200)).until(() -> !disposable1.isDisposed());
+        final Disposable disposable2 = flux.subscribe();
+        await().untilAsserted(() -> {
+            assertThat(disposable1.isDisposed()).isTrue();
+            assertThat(disposable2.isDisposed()).isTrue();
+        });
+    }
+
+    @Test
+    void subscriberContextIsNotMissing() {
+        final ClientRequestContext ctx = newContext();
+        final Flux<String> flux;
+
+        flux = Flux.deferContextual(reactorCtx -> {
+            assertThat((String) reactorCtx.get("foo")).isEqualTo("bar");
+            return Flux.just("baz");
+        });
+
+        final Flux<String> flux1 = flux.contextWrite(reactorCtx -> reactorCtx.put("foo", "bar"));
+        StepVerifier.create(flux1, initialReactorContext(ctx))
+                    .expectSubscriptionMatches(s -> ctxExists(ctx))
+                    .expectNextMatches(s -> ctxExists(ctx) && "baz".equals(s))
+                    .verifyComplete();
+    }
+
+    @Test
+    void ctxShouldBeCleanUpEvenIfErrorOccursDuringReactorOperationOnSchedulerThread() {
+        // Given
+        final ClientRequestContext ctx = newContext();
+        final Flux<String> flux;
+
+        // When
+        flux = Flux.just("Hello", "Hi")
+                   .delayElements(Duration.ofMillis(1000))
+                   .map(s -> {
+                       if (s.equals("Hello")) {
+                           throw new RuntimeException();
+                       }
+                       return s;
+                   })
+                   .contextWrite(Context.of(RequestContextAccessor.accessorKey(), ctx));
+
+        // Then
+        StepVerifier.create(flux)
+                    .expectError(RuntimeException.class)
+                    .verify();
+
+        assertThat(ctxExists(ctx)).isFalse();
+    }
+
+    @Test
+    void ctxShouldBeCleanUpEvenIfErrorOccursDuringReactorOperationOnMainThread() {
+        // Given
+        final ClientRequestContext ctx = newContext();
+        final Flux<String> flux;
+
+        // When
+        flux = Flux.just("Hello", "Hi")
+                   .map(s -> {
+                       if (s.equals("Hello")) {
+                           throw new RuntimeException();
+                       }
+                       return s;
+                   })
+                   .contextWrite(Context.of(RequestContextAccessor.accessorKey(), ctx));
+
+        // Then
+        StepVerifier.create(flux)
+                    .expectError(RuntimeException.class)
+                    .verify();
+
+        assertThat(ctxExists(ctx)).isFalse();
```

Comment:
ditto

Reply:
@minwoox nim, this code is different with unit test above (`ctxShouldBeCleanUpEvenIfErrorOccursDuringReactorOperationOnMainThread`)

Both `Reactive Stream` and `StepVerifier` are operated on `Main Thread`. 
Thus, if there is `ThreadLocal` pollution due to error during `Reactor Operation`,  `assertThat(ctxExists(ctx))` is should be `True`.